File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)

{
    float change;
    int cents;

    //Ensure validity of change
    do
    {
        change = get_float("Change owed: ");
    }
    while (change < 0);

    // Converting change to cents and rounding off
    change = cents = round(change * 100);

    int i = 0;
    //Quarters (0.25$)
    while (cents >= 25)
    {
        i++;
        cents -= 25;
    }
    //Dimes (0.10$)
    while (cents >= 10)
    {
        i++;
        cents -= 10;
    }
    //Nickels (0.05$)
    while (cents >= 5)
    {
        i++;
        cents -= 5;
    }
    //Pennies (0.01$)
    while (cents >= 1)
    {
        i++;
        cents -= 1;
    }
    printf("Coins used: %i\n", i);
}

File: pset6/sentimentals/cash.py
from cs50 import get_float

def main():

    # Prompt user for change owed
    while True:
        change = get_float("Change owed: ")

        # Ensure change is >= 0
        if change >= 0:
            break

    # Rounding off change to cents
    cents = round(change * 100)

    # Initiailize coins
    coins = 0

    # Quarter (0.25$)
    coins += cents // 25
    cents %= 25

    # Dimes (0.10$)
    coins += cents // 10
    cents %= 10

    # Nickels (0.05$)
    coins += cents // 5
    cents %= 5

    # Pennies (0.01$)
    coins += cents // 1
    cents %= 1

    # Print no. of coins used
    print(coins)

if __name__ == "__main__":
    main()



